From caea7a2c3d45b133d50a04633b57efe02bea1dc4 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 08 Aug 2021 18:25:08 +0300
Subject: [PATCH] [SW-0] gaudi2: arm: add address space comments

Gaudi2 address space is seen by ARM not the same way as it is in Greco
or Gaudi1.
This brings different offsets and addresses to Gaudi2 code which are
also differ from specs and architecture documents.
Add comments explaining the address map as seen by ARM.

Change-Id: I3fef3daa587f446f1a7323a034558de9d069fd1a
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common.h b/arch/gaudi2/common.h
index 53c1868..c4341fd 100644
--- a/arch/gaudi2/common.h
+++ b/arch/gaudi2/common.h
@@ -9,6 +9,15 @@
 #include <gaudi2_blocks_linux_driver.h>
 #include <bitops.h>
 
+/*
+ * Gaudi2 ARM address extension is defined by bits 38,39 in the address and
+ * in the below way:
+ * 0b00	- CFG Access	(0x3FFxxxxxxx)
+ * 0b01	- SRAM Access	(0x7FFDxxxxxx)
+ * 0b10	- HBM Access	(0x8xxxxxxxxx)
+ * 0b11	- PCIe Access	(0xCxxxxxxxxx)
+ */
+
 #ifdef TARGET_IS_ONIX
 #define PPBOOT_TEXT_BASE		0x8200000000
 #define SDRAM_BASE			0x8200000000ull
@@ -30,6 +39,12 @@
 #define BAR0_RSRVD_SIZE			0x1000000
 
 #define HL_SPI_BASE			0x3FF6000000ull
+/*
+ * The actual Gaudi2 CFG space starts at 0x7FFC000000ull, but
+ * all the specs offsets were generated against the new DBG space base, which
+ * is 0x7FF8000000ull, so to be consistent, we define CFG base to start at the
+ * DBG base.
+ */
 #define HL_CFG_BASE			0x3FF8000000ull
 #define STM_FLASH_BASE			0x1000007FF4000000ull
 #define BAR0_RSRVD_BASE			0x1000FFFFFC000000ull
